mm2src/adex_cli/src/scenarios/init_mm2_cfg.rs

```diff
@@ -0,0 +1,302 @@
+            .with_help_message(r#"Port to use for RPC communication. Optional, defaults to 7783"#)
+            .with_placeholder(DEFAULT_OPTION_PLACEHOLDER)
+            .prompt()
+            .map_err(|error| {
+                error!("Failed to get rpcport: {error}");
+            })?
+            .into();
+        Ok(())
+    }
+
+    fn inquire_rpc_local_only(&mut self) -> Result<(), ()> {
+        self.rpc_local_only = CustomType::<InquireOption<bool>>::new("What is rpc_local_only:")
+                .with_parser(OPTION_BOOL_PARSER)
+                .with_formatter(DEFAULT_OPTION_BOOL_FORMATTER)
+                .with_default_value_formatter(DEFAULT_DEFAULT_OPTION_BOOL_FORMATTER)
+                .with_default(InquireOption::None)
+                .with_help_message("If false the AtomicDEX API will allow rpc methods sent from external IP addresses. Optional, defaults to true. Warning: Only use this if you know what you are doing, and have put the appropriate security measures in place.")
+                .prompt()
+                .map_err(|error| {
+                    error!("Failed to get rpc_local_only: {error}");
+                })?.into();
+        Ok(())
+    }
+
+    fn inquire_i_am_a_seed(&mut self) -> Result<(), ()> {
+        self.i_am_seed = CustomType::<InquireOption<bool>>::new("What is i_am_a_seed:")
+                .with_parser(OPTION_BOOL_PARSER)
+                .with_formatter(DEFAULT_OPTION_BOOL_FORMATTER)
+                .with_default_value_formatter(DEFAULT_DEFAULT_OPTION_BOOL_FORMATTER)
+                .with_default(InquireOption::None)
+                .with_help_message("Runs AtomicDEX API as a seed node mode (acting as a relay for AtomicDEX API clients). Optional, defaults to false. Use of this mode is not reccomended on the main network (7777) as it could result in a pubkey ban if non-compliant. on alternative testing or private networks, at least one seed node is required to relay information to other AtomicDEX API clients using the same netID.")
+                .prompt()
+                .map_err(|error| {
+                    error!("Failed to get i_am_a_seed: {error}");
+                })?.into();
+        Ok(())
+    }
+
+    fn inquire_seednodes(&mut self) -> Result<(), ()> {
+        info!("Reading seed nodes until tap enter is met");
+        loop {
+            let seednode: Option<Ipv4Addr> = CustomType::<InquireOption<Ipv4Addr>>::new("What is next seednode:")
+                  .with_help_message("Optional. If operating on a test or private netID, the IP address of at least one seed node is required (on the main network, these are already hardcoded)")
+                  .with_placeholder(DEFAULT_OPTION_PLACEHOLDER)
+                  .prompt()
+                  .map_err(|error| {
+                      error!("Failed to get seed node: {error}");
+                  })?.into();
+            if seednode.is_none() {
```

Comment:
Hello, @shamardy! Thank you for your feedback!

There are some concerns regarding whether or not we should provide default values in `adex`, or let `mm2` decide which defaults to use.

You are correct, except for the fact that `mm2` already has functionality to set default configuration values.

It is worth noting that using default configurations, rather than hardcoding them, would be preferable. This applies not only to default seed nodes, but also to other values such as electrum connection timeout.

In the current circumstances, if some configuration parts are missing, `mm2` is responsible for setting up defaults. This is a single point of responsibility and quite sensitive. If we were to change the behavior in `mm2`, someone would have to pay attention to Adex and make changes as well.

In this situation, we should not spread this responsibility between `adex` and `mm2`. Instead, we should let `adex` set nothing as the default and let MM2 keep and implement this responsibility.

Once we get rid of the hardcoded functionality, I would be happy to let `adex` take charge.

Reply:
What do you think?

Reply:
You are right, for seednodes the cli doesn't need a default value since when running the binary it will be set by mm2 if `netid` is 7777. I got confused a bit when reviewing this part before looking at how cli will run mm2 :)